fuel-vm/src/call.rs

```diff
@@ -124,43 +123,53 @@ impl CallFrame {
         to: ContractId,
         asset_id: AssetId,
         registers: [Word; VM_REGISTER_COUNT],
+        code_size: Word,
         a: Word,
         b: Word,
-        code: Contract,
     ) -> Self {
         Self {
             to,
             asset_id,
             registers,
+            code_size,
             a,
             b,
-            code,
         }
     }
 
-    /// Contract code of the called (`to`) id.
-    pub fn code(&self) -> &[u8] {
-        self.code.as_ref()
+    /// Start of the contract id offset from the beginning of the call frame.
+    pub const fn contract_id_offset() -> usize {
+        0
     }
 
-    /// Contract code memory offset.
-    pub const fn code_offset() -> usize {
-        Self::code_size_offset() + WORD_SIZE
+    /// Start of the asset id offset from the beginning of the call frame.
+    pub const fn asset_id_offset() -> usize {
+        ContractId::LEN
```

Comment:
Perhaps something like this would make the layout relationship a bit more clear and automatic: 
```suggestion
        Self::contract_id_offset() + ContractId::LEN
```

Reply:
I'm not sure I follow? The asset id offset is one `ContractId::LEN` from the start. Or do you mean `Self::contract_id_offset()` returns `0`?

Reply:
Yeah, exactly, it looks like contract id is at 0, and asset id comes right after contract id. So it makes sense that it would be one `ContractId::LEN`.  but if we ever change `contract_id_offset`, then including it in the `asset_id_offset` calculation will ensure that it is still in the right position.